### ISSUE TITLE
columnApi.autoSizeColumn(...) is not actionable at first-render.

### ISSUE THREAD

**Issue by harish-prakash** [ag-grid/ag-grid#6655]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**
There is NO WAY! to guarantee `.autoSizeColumn(...)` will be applied when the grid is ready (i.e. presented to the user on the browser). The only HACKY work-around is to use `setTimeout(() => columnApi.autoSizeColumn(...))` which doesn't necessarily guarantee, just increases the probability 🤦‍♂️

**Expected behavior**
There must be an event that WILL BE FIRED when the grid is ready for initial display enhancements.

**Details**
* **Problem Description**: `onFirstDataRendered` is often fired before `.autoSizeColumn(...)` can be called.
* **AgGrid version**: 29.2.0
* **CodeSandbox**: https://codesandbox.io/s/sad-shirley-vq9g2q?file=/src/App.tsx
* **Framework**: React
* **Language**: TypeScript
* **Misleading or incomplete documentation**: https://www.ag-grid.com/react-data-grid/column-sizing/#resizing-columns-when-data-is-rendered
* **Related Tickets**: https://github.com/ag-grid/ag-grid/issues/3263

The above ag-grid issue clearly states that `onFirstDataRendered` is an (quote) "internal rendering" state-triggered-event and not a display render event. Then, **here are the problems**
* Rename the event to appropriately reflect its function, e.g. `onFirstDataLoaded` or `onFirstDataInternallyRendered`.
* Give a guaranteed way to use `.autoSizeColumn(...)`

The bug doesn't present it self always, here is a clip where I had to refresh dozens of times before `autoSizing` was correctly applied (every refresh after that did not apply). The success:failure ratio is 1-success:12-failures at best.

> https://github.com/ag-grid/ag-grid/assets/32839065/af451e93-0bf1-43ba-a59b-95f1eb1aa8f0

**Comment by kiril-matev** [comment]
Hello @harish-prakash,

Thank you for reporting this.

Please note your example is using AG Grid v29.2. We've made a number of changes to better support asynchronous rendering in React 18 since and released them as part of AG Grid v30.

I've upgraded your original example to AG Grid v30. Please note the issue you reported isn't reproducible in this upgraded example anymore:
https://codesandbox.io/s/prod-cache-8nwrmc?file=/package.json

You can also see a better example to demonstrate this here:
https://codesandbox.io/s/happy-khayyam-qw65yd?file=/package.json
Note the AGE column has a large initial width value set which is then modified by the auto-size logic in the onFirstDataRendered event handler.

Please verify this behavior with the examples above and upgrade to the latest version of AG Grid.

**Comment by harish-prakash** [comment]
Will test this out and provide my feedback, thank you 🫡

**Comment by zoheil-ag** [comment]
Hi @harish-prakash,

As it's been a while since the last interaction on this issue, I will be closing this as resolved. Feel free to let us know if the problem still occurs for you.

Kind regards,

**Comment by zoheil-ag** [comment]
Hi @randdusing,

You can search our [changelog](https://ag-grid.com/changelog/) with keywords to find anything you are keen on in particular. In regards to this specifically, I believe **AG-8399** (released in 29.3.2) would be most applicable. 

Please note there may be other changes we've made that have generally improved this area.

Hope that helps.

Kind regards,
Zoheil
